mesonbuild/backend/backends.py

```diff
@@ -1429,7 +1429,19 @@ def generate_target_install(self, d: InstallData) -> None:
                         d.targets.append(i)
 
     def generate_custom_install_script(self, d: InstallData) -> None:
-        d.install_scripts = self.build.install_scripts
+        result: T.List[ExecutableSerialisation] = []
+        srcdir = self.environment.get_source_dir()
+        builddir = self.environment.get_build_dir()
+        for i in self.build.install_scripts:
+            fixed_args = []
+            for a in i.cmd_args:
+                a = a.replace('@SOURCE_ROOT@', srcdir)
+                a = a.replace('@BUILD_ROOT@', builddir)
+                fixed_args.append(a)
+            es = copy.copy(i)
```

Comment:
The `import copy` needs to be restored, as well.

Reply:
You're right. Changed my implementation now.